build/common/installer/scripts/tomlparser-hostlogs-geneva-config.rb

```diff
@@ -0,0 +1,145 @@
+#!/usr/local/bin/ruby
+
+require "tomlrb"
+require "json"
+
+require_relative "ConfigParseErrorLogger"
+
+@configMapMountPath = "./etc/config/settings/hostlogs-settings"
+@configSchemaVersion = ""
+
+# configmap settings related to geneva logs config
+GENEVA_SUPPORTED_ENVIRONMENTS = ["Test", "Stage", "DiagnosticsProd", "FirstpartyProd", "BillingProd", "ExternalProd", "CaMooncake", "CaFairfax", "CaBlackforest"]
+@geneva_account_environment = ""
+@geneva_account_name = ""
+@geneva_account_namespace = ""
+@geneva_logs_config_version = ""
+@geneva_gcs_authid = ""
+@azure_json_path = "C:\\k\\azure.json"
+
+# Use parser to parse the configmap toml file to a ruby structure
+def parseConfigMap
+  begin
+    # Check to see if config map is created
+    if (File.file?(@configMapMountPath))
+      puts "config::configmap container-azm-ms-hostlogsconfig for agent settings mounted, parsing values"
+      parsedConfig = Tomlrb.load_file(@configMapMountPath, symbolize_keys: true)
+      puts "config::Successfully parsed mounted config map"
+      return parsedConfig
+    else
+      puts "config::configmap container-azm-ms-hostlogsconfig for agent settings not mounted, using defaults"
+      return nil
+    end
+  rescue => errorStr
+    ConfigParseErrorLogger.logError("Exception while parsing config map for agent settings : #{errorStr}, using defaults, please check config map for errors")
+    return nil
+  end
+end
+
+# Use the ruby structure created after config parsing to set the right values to be used as environment variables
+def populateSettingValuesFromConfigMap(parsedConfig)
+  begin
+    if !parsedConfig.nil? && !parsedConfig[:hostlogs_settings].nil?
+      geneva_logs_config = parsedConfig[:hostlogs_settings][:geneva_logs_config]
+      if !geneva_logs_config.nil?
+        puts "config: parsing geneva_logs_config settings"
+        geneva_account_environment = geneva_logs_config[:environment]
+        geneva_account_name = geneva_logs_config[:account]
+        geneva_account_namespace = geneva_logs_config[:namespace]
+        geneva_logs_config_version = geneva_logs_config[:configversion]
+        if !geneva_account_environment.nil? && !geneva_account_name.nil? && !geneva_account_namespace.nil? && !geneva_logs_config_version.nil?
+          if GENEVA_SUPPORTED_ENVIRONMENTS.include?(geneva_account_environment)
```

Comment:
Either we need to extend this map for airgap clouds or remove this check. Based on my discussion with geneva team, they recommend to skip the validation on the environment name as customer knows these values based on the internal doc.

Reply:
Sounds good.